libs/permit-utils/src/utils/getTokenName.ts

```diff
@@ -0,0 +1,14 @@
+import type { JsonRpcProvider } from '@ethersproject/providers'
+
+import { getAddress } from '@ethersproject/address'
+
+import { getContract } from './getContract'
+
+import Erc20Abi from '../abi/erc20.json'
+
+export async function getTokenName(tokenAddress: string, chainId: number, provider: JsonRpcProvider): Promise<string> {
+  const formattedAddress = getAddress(tokenAddress)
+  const erc20Contract = getContract(formattedAddress, Erc20Abi, provider)
+
+  return erc20Contract.callStatic['name']()
```

Comment:
If you set a generic type for `Contract` you will be able to call methods like:
```ts
erc20Contract.callStatic.name()
```

Reply:
Didn't work
![image](https://github.com/cowprotocol/cowswap/assets/43217/c94571a0-659c-4fd3-909d-47aa508cf91a)

Or do you mean to do it some other way?